draft-pauly-add-requirements.md

```diff
@@ -186,68 +186,73 @@ Using a content-provider's encrypted resolver can also provide several privacy a
 and the content provider, assuming that the content provider would already see the names upon a secure connection
 later being made based on the DNS answers (e.g., in the TLS SNI extension)
 
-# Distilled discovery mechanisms {#mechanisms}
+# Discovery mechanisms {#mechanisms}
 
-The various use cases described in {{use-cases}} do not all necessarily require separate mechanisms.
+The use cases described in {{use-cases}} do not all necessarily require separate mechanisms.
 
-Generally, the use cases can be summarized in two catgories:
+Generally, the use cases can be summarized in two categories:
 
-1. Discover encrypted resolvers equivalent to (or associated with) unencrypted resolvers. (Network-provisioned,
-client-selected, VPN)
-2. Discover encrypted resolvers applicability to a limited set of domains. (Private names, local content, CDN content)
+1. Encrypted resolver discovery: Discover encrypted resolvers equivalent to (or associated with) unencrypted resolvers.
+Examples include network-provisioned, client-selected, and VPN-configured resolvers.
+2. Domain-specific encrypted resolver discovery: Discover encrypted resolvers applicable to a limited set of domains.
+Examples include resolvers for enterprise or private names, local content, and CDN content.
 
-For the first group, mechanisms to discover equivalent resolvers can either add new parameters to existing provisioning
+Encrypted resolver discovery mechanisms can either add new parameters to existing provisioning
 mechanisms (adding necessary information to use DoT or DoH to options in DHCP, RAs, or IKEv2) or else provide a way
 to communicate with a provisioned unencrypted DNS resolver and discover the equivalent or associated encrypted
 DNS resolver.
 
-For the second group, mechanisms additionally need to provide some information about the applicability and capabilities
-of encrypted resolvers. This information could either be provisioned,
+Domain-specific encrypted resolver discovery mechanisms additionally need to provide some information about the
+applicability and capabilities of encrypted resolvers. This information could either be provisioned,
 or be discovered based on clients actively trying to access content.
 
 # Privacy and security requirements {#priv-sec}
 
-Since one of the main reasons for enabling encrypted DNS on clients is to improve the privacy and security properties
-of DNS queries, discovery mechanisms must be designed carefully to not weaken a client's security or privacy.
+A primary goal of encrypted DNS is improving the privacy and security of DNS queries and answers in the presence
+of malicious attackers. Such attackers are assumed to interfere with or otherwise impede DNS traffic and corresponding
```

Comment:
Is this the right attacker to focus on? I think there are a couple different attacks:
- Attacker (may be off-path) monitors you by observing your unencrypted DNS traffic to profile you
- On-path attacker modifies your unencrypted DNS traffic to mess with you
- On-path attacker blocks encrypted DNS connections

Attacks about redirecting encrypted DNS traffic might actually be the second case—not blocking, but changing.

Reply:
What I tried to capture was an adversary whose capabilities encompass all of the above: it is everywhere (on-path and off-path) and active. It's true that the attacker may have different goals given these capabilities. Let me take another cut at that.